contracts/staking/contracts/src/stake/MixinStakeStorage.sol

```diff
@@ -0,0 +1,196 @@
+    ///      This causes both the `current` and `next` fields to increase immediately.
+    /// @param balancePtr storage pointer to balance.
+    /// @param amount to mint.
+    function _mintBalance(IStructs.DelayedBalance storage balancePtr, uint256 amount)
+        internal
+    {
+        // Remove stake from balance
+        IStructs.DelayedBalance memory balance = _syncBalanceDestructive(balancePtr);
+        balance.next = LibSafeDowncast.downcastToUint96(uint256(balance.next).safeAdd(amount));
+        balance.current = LibSafeDowncast.downcastToUint96(uint256(balance.current).safeAdd(amount));
+
+        // update state
+        _storeBalance(balancePtr, balance);
+    }
+
+    /// @dev Burns existing value in a balance.
+    ///      This causes both the `current` and `next` fields to decrease immediately.
+    /// @param balancePtr storage pointer to balance.
+    /// @param amount to mint.
+    function _burnBalance(IStructs.DelayedBalance storage balancePtr, uint256 amount)
+        internal
+    {
+        // Remove stake from balance
+        IStructs.DelayedBalance memory balance = _syncBalanceDestructive(balancePtr);
+        balance.next = LibSafeDowncast.downcastToUint96(uint256(balance.next).safeSub(amount));
+        balance.current = LibSafeDowncast.downcastToUint96(uint256(balance.current).safeSub(amount));
+
+        // update state
+        _storeBalance(balancePtr, balance);
+    }
+
+    /// @dev Increments a balance.
+    ///      Ths updates the `next` field but not the `current` field.
+    /// @param balancePtr storage pointer to balance.
+    /// @param amount to increment by.
+    function _incrementBalance(IStructs.DelayedBalance storage balancePtr, uint256 amount)
+        internal
+    {
+        // Add stake to balance
+        IStructs.DelayedBalance memory balance = _syncBalanceDestructive(balancePtr);
+        balance.next = LibSafeDowncast.downcastToUint96(uint256(balance.next).safeAdd(amount));
+
+        // update state
+        _storeBalance(balancePtr, balance);
+    }
+
+    /// @dev Decrements a balance.
+    ///      Ths updates the `next` field but not the `current` field.
+    /// @param balancePtr storage pointer to balance.
+    /// @param amount to decrement by.
+    function _decrementBalance(IStructs.DelayedBalance storage balancePtr, uint256 amount)
+        internal
+    {
+        // Remove stake from balance
+        IStructs.DelayedBalance memory balance = _syncBalanceDestructive(balancePtr);
+        balance.next = LibSafeDowncast.downcastToUint96(uint256(balance.next).safeSub(amount));
+
+        // update state
+        _storeBalance(balancePtr, balance);
+    }
+
+    /// @dev Stores a balance in storage.
+    /// @param balancePtr points to where `balance` will be stored.
+    /// @param balance to save to storage.
+    function _storeBalance(
+        IStructs.DelayedBalance storage balancePtr,
+        IStructs.DelayedBalance memory balance
+    )
+        private
```

Comment:
Nit: Same as above re testing

Reply:
Yeah, I think you're right. For unit testing these will have to be made `internal` =\